web_widget_x2many_2d_matrix/README.rst

```diff
@@ -51,12 +55,42 @@ show_row_totals
 show_column_totals
     If field_value is a numeric field, calculate column totals
 
+Example
+=======
+
+You need a data structure already filled with values. Let's assume we want to use this widget in a wizard that lets the user fill in planned hours for one task per project per user. In this case, we can use ``project.task`` as our data model and point to it from our wizard. The crucial part is that we fill the field in the default function::
+
+ class MyWizard(models.TransientModel):
+    _name = 'my.wizard'
+
+    def _default_task_ids(self):
+        # your list of project should come from the context, some selection
+        # in a previous wizard or wherever else
+        projects = self.env['project.project'].browse([1, 2, 3])
+        # same with users
+        users = self.env['res.users'].browse([1, 2, 3])
+        return [
+            (0, 0, {'project_id': p.id, 'user_id': u.id, 'planned_hours': 0})
+            # if the project doesn't have a task for the user, create a new one
+            if not p.task_ids.filtered(lambda x: x.user_id == u) else
+            # otherwise, return the task
+            (4, p.task_ids.filtered(lambda x: x.user_id == u)[0].id)
+            for p in projects
+            for u in users
+        ]
+
+    task_ids = fields.Many2many('project.task', default=_default_task_ids)
+
+Now in our wizard, we can use::
+
+<field name="task_ids" widget="x2many_2d_matrix" field_x_axis="project_id" field_y_axis="user_id" field_value="planned_hours" />
+
 Known issues / Roadmap
 ======================
 
 * it would be worth trying to instantiate the proper field widget and let it render the input
 
-
+===========
```

Comment:
These sections should the line on top. It's only for the first title to highlight it